lib/grppccom.gi

```diff
@@ -1207,15 +1207,51 @@ end);
 #M  ComplementcClassesRepresentatives( <G>, <N> )
 ##
 InstallMethod( ComplementClassesRepresentatives,
-  "tell that the normal subgroup must be solvable",IsIdenticalObj,
-  [IsGroup,IsGroup],-2*RankFilter(IsGroup),
+  "tell that the normal subgroup or factor must be solvable", IsIdenticalObj,
+  [ IsGroup, IsGroup ], -2*RankFilter(IsGroup),
 function( G, N )
-  if IsSolvableGroup(N) then
+  if IsSolvableGroup(N) or HasSolvableFactorGroup(G, N) then
     TryNextMethod();
   fi;
-  Error("Cannot compute complement classes for nonsolvable normal subgroups");
+  Error("Cannot compute complements if both N and G/N are nonsolvable");
 end);
 
+
+#############################################################################
+##
+#M  ComplementcClassesRepresentatives( <G>, <N> ) . from conj. cl. subgroups
+##
+InstallMethod( ComplementClassesRepresentatives,
+  "using conjugacy classes of subgroups", IsIdenticalObj,
+  [ IsGroup and HasConjugacyClassesSubgroups, IsGroup ], 0,
+
+function( G, N )
+
+  local C, Hc, H;
+
+  # if <N> is trivial the only complement is <G>
+  if IsTrivial(N) then
+      C := [ G ];
+  # if <G> and <N> are equal the only complement is trivial
+  elif G = N  then
+      C := [ TrivialSubgroup(G) ];
+  elif not IsNormal(G, N) then
+    Error("N must be normal in G");
+  else
+    C := [ ];
+    for Hc in ConjugacyClassesSubgroups(G) do
+      H := Representative(Hc);
+      if (( CanComputeSize(G) and CanComputeSize(N) and CanComputeSize(H) and
+	  Size(G) = Size(N)*Size(H) ) or G = ClosureGroup(N, H) )
+	  and IsTrivial(Intersection(N, H)) then
+	Add(C, H);
+      fi;
+    od;
+  fi;
+  return C;
+end);
+
```

Comment:
Looks fine to me. Of course it is a "naive" method and could have horrible performance, but I tend to agree with @markuspf that this is still better than an error. Regarding the concern that this "suboptimal" algorithm could tarnish GAP's reputation, I say two things: First off, let's then just add a brief comment here that explains this: "This is a very naiv algorithm; we only provide it as a last resort, ideally it is never invoked". Secondly: if we are concerned about that, there is a lot of code in GAP we should rip out. There are far too many times when trying to do a seemingly simple operation, I cought GAP trying to enumerate a huge or even infinite set. Even things like trying to compute the normalizer of the trivial group can run into an infinite computation in GAP -- for perfectly fine reasons, of course, but still, this seems far worse to me than the above method... And the above method in turn seems much better to me than saying "oops, don't how to do that" when people ask GAP to do this for tivial examples (like G = A5xA5)